docs/migration/00_migration_guide_to_v0.5.ipynb

```diff
@@ -0,0 +1,562 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Qiskit Optimization v0.5 Migration Guide\n",
+    "\n",
+    "This tutorial will guide you through the process of migrating your code from Qiskit Optimization v0.4 to v0.5."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Overview\n",
+    "\n",
+    "Qiskit Terra v0.22 introduces new algorithm implementations that leverage [Qiskit Primitives](https://qiskit.org/documentation/apidoc/primitives.html) (Estimator and Sampler). The former algorithm implementations that leverage opflow will be deprecated in the future release.\n",
+    "\n",
+    "Qiskit Optimization v0.5 supports both the new and the former algorithms of Qiskit Terra v0.22 until the former algorithms are deprecated.\n",
+    "\n",
+    "It is not the intention to provide detailed explanations of the primitives in this migration guide. We suggest that you read the [corresponding resources](https://qiskit.org/documentation/apidoc/primitives.html) of the Qiskit Terra documentation instead.\n",
+    "\n",
+    "We use `qiskit.primitives.Sampler` in this guide as an example of Sampler implementation, which follows `qiskit.primitives.BaseSampler` interface. Users can also use other Sampler implementations such as `BackendSampler` (qiskit-terra), `AerSampler` (qiskit-aer), and Qiskit Runtime Sampler (qiskit-ibm-runtime)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## `MinimumEigenOptimizer`\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The former algorithms exist in `qiskit.algorithms.minimum_eigen_solvers` and we can access them by `qiskit.algorithms.*`. On the other hand, the new algorithms exist in `qiskit.algorithms.minimum_eigensolvers` and we can access them by `qiskit.algorithms.minimum_eigensolvers.*`. Note that the difference is `minimum_eigen_solvers` (former) and `minimum_eigensolvers` (new).\n",
+    "\n",
+    "`MinimumEigenOptimizer` of Qiskit Optimization can use `qiskit.algorithms.MinimumEigenSolver` interface of the former algorithms and \n",
+    "`qiskit.algorithms.minimum_eigensolvers.SamplingMinimumEigensolver` interface of the new algorithms.\n",
+    "Note that `MinimumEigenOptimizer` cannot basically handle `qiskit.algorithms.minimum_eigensolvers.MinimumEigensolver` of the new algorithms.\n",
```

Comment:
I was thinking about this... If a user currently passes a new MinimumEigensolver (except for the Numpy one), say VQE, then since the methods of that type are the same as in SamplingMinimumEigensolver I think the MinimumEigenOptimzer will not fail until it attempts to access a field in the result after the (say new VQE) algorithm has run, i,e eigenstate. This means it may waste a whole bunch of compute before the mismatch is detected. I am wondering if the MinimumEigenOptimizer should check if its passed a new MinimumEigensolver and raise an error if its not the Numpy one - otherwise a user may end up sending a lot of jobs to a device to run the algorithm only for it to fail right at the end.

Reply:
Good catch. I will update `MinimumEigenOptimizer` to raise an error if non-compatible eigensolver is given.

Reply:
I added error messages #450